core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -543,6 +543,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    if (contains("spark.cores.max")) {
+      val totalCores = getInt("spark.cores.max", -1)
```

Comment:
Please use `Option` instead, I think we don't need to check them if they're not set. Also looks like if we don't set this configuration, will it always lead to exception as default value is -1.